### PR TITLE
Add Consuming Avro messages from Kafka using the schema-registry exam…

### PR DESCRIPTION
**Pull request by ppalaga**
…ple to Kafka component docs

This follows up the discussion in https://github.com/apache/camel-quarkus/issues/1960 .

Could you please review and perhaps also add more details @tstuber?

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc [hunk 1]
```diff
@@ -580,3 +580,23 @@ from("kafka:my_topic?headerFilterStrategy=#myStrategy")
...
+
+This route applies the schema and converts the Avro messages to POJOs for further processing:
+
+[source,java]
+----
+from(kafka("avro-events")
+    .brokers("localhost:9092")
+    .schemaRegistryURL("http://localhost:8081")
+    .specificAvroReader(true)
+    .valueDeserializer("io.confluent.kafka.serializers.KafkaAvroDeserializer")
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc:595]: Is it only confluent that has avra deseriazlier and this implies you are using confluent schema registry. Can we make this example also work with other schema registries? And add some more notes on this? And btw the example is using endpoint-dsl which we should mention.

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc [hunk 2]
```diff
@@ -580,3 +580,23 @@ from("kafka:my_topic?headerFilterStrategy=#myStrategy")
...
+    .resolve(getContext()))
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc:596]: You would not call resolve when using endpoint-dsl creating routes

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc (hunk 1 above)
Comment (ppalaga) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc:595]: I am all for improving it, but please note that I am totally lacking on the required expertise. All I did in this PR is just summing up the info gathered by @tstuber in https://github.com/apache/camel-quarkus/issues/1960 .

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc (hunk 2 above)
Comment (ppalaga) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc:596]: OK, let me remove resolve()

Comment (ppalaga) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc]: Fixed in d5154d9

**Review by oscerd — APPROVED** [review]
We can merge this one and then doing the same for Apicurio. We can note this is for Confluent Registry.

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc (hunk 1 above)
Comment (tstuber) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kafka-component.adoc:595]: Actually, the adjusted code misses now a closing bracket at the end: `.valueDeserializer("io.confluent.kafka.serializers.KafkaAvroDeserializer"))`

**Comment by tstuber** [comment]
I added a complete example of reading & writing to Kafka using avro format (and confluent schema registry): 
https://github.com/tstuber/camel-quarkus-kafka-schema-registry

@ppalaga does that help you for documenting? Otherwise, the changes in your PR look good to me and I think it will help other people

**Comment by oscerd** [comment]
Lets merge this and we'll do the same for other schema registries.

**Pull request merged**